crates/router/src/core/connector_validation.rs

```diff
@@ -0,0 +1,581 @@
+use api_models::enums as api_enums;
+use common_utils::pii;
+use error_stack::ResultExt;
+use external_services::http_client::client;
+use masking::PeekInterface;
+use pm_auth::connector::plaid::transformers::PlaidAuthType;
+
+use crate::{
+    core::errors::{self},
+    types,
+    types::transformers::ForeignTryFrom,
+};
+
+pub struct ConnectorAuthTypeAndMetadataValidation<'a> {
+    pub connector_name: &'a api_models::enums::Connector,
+    pub auth_type: &'a types::ConnectorAuthType,
+    pub connector_meta_data: &'a Option<pii::SecretSerdeValue>,
+}
+
+impl ConnectorAuthTypeAndMetadataValidation<'_> {
+    pub fn validate_auth_and_metadata_type(
+        &self,
+    ) -> Result<(), error_stack::Report<errors::ApiErrorResponse>> {
+        let connector_auth_type_validation = ConnectorAuthTypeValidation {
+            auth_type: self.auth_type,
+        };
+        connector_auth_type_validation.validate_connector_auth_type()?;
+        self.validate_auth_and_metadata_type_with_connector()
+            .map_err(|err| match *err.current_context() {
+                errors::ConnectorError::InvalidConnectorName => {
+                    err.change_context(errors::ApiErrorResponse::InvalidRequestData {
+                        message: "The connector name is invalid".to_string(),
+                    })
+                }
+                errors::ConnectorError::InvalidConnectorConfig { config: field_name } => err
+                    .change_context(errors::ApiErrorResponse::InvalidRequestData {
+                        message: format!("The {} is invalid", field_name),
+                    }),
+                errors::ConnectorError::FailedToObtainAuthType => {
+                    err.change_context(errors::ApiErrorResponse::InvalidRequestData {
+                        message: "The auth type is invalid for the connector".to_string(),
+                    })
+                }
+                _ => err.change_context(errors::ApiErrorResponse::InvalidRequestData {
+                    message: "The request body is invalid".to_string(),
+                }),
+            })
+    }
+
+    fn validate_auth_and_metadata_type_with_connector(
+        &self,
+    ) -> Result<(), error_stack::Report<errors::ConnectorError>> {
+        use crate::connector::*;
+
+        match self.connector_name {
+            api_enums::Connector::Vgs => {
+                vgs::transformers::VgsAuthType::try_from(self.auth_type)?;
+                Ok(())
+            }
+            api_enums::Connector::Adyenplatform => {
+                adyenplatform::transformers::AdyenplatformAuthType::try_from(self.auth_type)?;
+                Ok(())
+            }
+            // api_enums::Connector::Payone => {payone::transformers::PayoneAuthType::try_from(val)?;Ok(())} Added as a template code for future usage
```

Comment:
We seem to be handling this variant in line 353, can we remove this commented code?